test/remote/recovery_key_tests.js

```diff
@@ -0,0 +1,138 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const assert = require('insist')
+const config = require('../../config').getProperties()
+const crypto = require('crypto')
+const TestServer = require('../test_server')
+const Client = require('../client')()
+const hkdf = require('../../lib/crypto/hkdf')
+const Promise = require('bluebird')
+const jose = require('node-jose')
+
+describe('remote recovery keys', function () {
+  this.timeout(10000)
+
+  let server, client, email
+  const password = '(-.-)Zzz...'
+
+  let recoveryKeyId
+  let recoveryData
+  let keys
+  let keystore
+
+  function createRecoveryKey(uid, kB) {
+    const recoveryCode = crypto.randomBytes(16)
+
+    return Promise.all([
+      hkdf(recoveryCode, uid, 'fxa recovery fingerprint', 16),
+      hkdf(recoveryCode, uid, 'fxa recovery encrypt key', 32)
+    ])
```

Comment:
Since the auth-server doesn't care about the crypto details of the recovery key, I wonder if it's really worth doing all of this just for testing purposes.  An alternative would be to just use `crypto.randomBytes()` to create a mock `recoveryKeyId` and `recoveryKey`, which would simplify the code here and would also avoid pulling in a dependency on node-jose for this repo.